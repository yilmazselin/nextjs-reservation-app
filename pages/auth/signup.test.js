import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import "jest-styled-components";
import { useSession } from "next-auth/react";
import { useTranslation } from "react-i18next";
import Signup from "./signup";

jest.mock("next-auth/react");
jest.mock("react-i18next", () => ({
  useTranslation: jest.fn(),
}));

const tSpy = jest.fn((str) => str);

useTranslation.mockReturnValue({
  t: tSpy,
  i18n: {
    changeLanguage: () => new Promise(() => {}),
  },
});

describe("Signup page test", () => {
  it("Should render signup page correctly", async () => {
    const mockSession = undefined;

    useSession.mockReturnValueOnce([mockSession, false]);
    const { container } = render(<Signup />);

    expect(container).toMatchSnapshot();

    const pageTitle = screen.getByText(/auth:signup-title/i);
    const emailInput = screen.getByLabelText(/common:labels.email/i);
    const passwordInput = screen.getByLabelText(/common:labels.password/i);
    const localeSelect = screen.getByLabelText(/common:labels.locale/i);
    const submitButton = screen.getByRole("submit-button");
    const tabBarListButton = screen.getByText(/common:tabbar.list/i);
    const tabBarAccountButton = screen.getByText(/common:tabbar.account/i);

    expect(pageTitle).toBeInTheDocument();
    expect(emailInput).toBeInTheDocument();
    expect(passwordInput).toBeInTheDocument();
    expect(localeSelect).toBeInTheDocument();
    expect(submitButton).toBeInTheDocument();
    expect(submitButton).toBeDisabled();
    expect(tabBarListButton).toBeInTheDocument();
    expect(tabBarAccountButton).toBeInTheDocument();
  });
});
