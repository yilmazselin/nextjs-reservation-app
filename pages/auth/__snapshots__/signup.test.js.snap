// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup page test Should render signup page correctly 1`] = `
.c3 {
  color: #121212;
  font-size: 32px;
  font-family: "NunitoSans-600","sans-serif";
  margin-top: 92px;
  margin-bottom: 40px;
}

.c15 {
  color: #121212;
  font-size: 12px;
  font-family: "NunitoSans-600","sans-serif";
  margin-top: 8px;
}

.c1 {
  width: 100%;
  margin-left: auto;
  margin-right: auto;
  padding-left: 24px;
  padding-right: 24px;
}

.c0 {
  padding-bottom: 100px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  margin-bottom: 40px;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  margin-bottom: 45px;
}

.c7 {
  color: #64738C;
  font-size: 16px;
  font-family: "NunitoSans-500","sans-serif";
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  pointer-events: none;
  -webkit-transform-origin: left top;
  -ms-transform-origin: left top;
  transform-origin: left top;
  -webkit-transform: scale(1) translate3d(0,22px,0);
  -ms-transform: scale(1) translate3d(0,22px,0);
  transform: scale(1) translate3d(0,22px,0);
  -webkit-transition: 200ms ease all;
  transition: 200ms ease all;
  position: relative;
}

.c5 {
  position: relative;
  border-radius: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #121212;
  font-size: 16px;
  font-family: "NunitoSans-600","sans-serif";
  line-height: 24px;
  padding-bottom: 8px;
  outline: 0;
  border: 0;
  border-bottom: 1px solid;
  border-bottom-color: #BBC3CF;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
}

.c5:focus ~ .c6,
.c5:not(:placeholder-shown) ~ .c6 {
  color: #77869e;
  -webkit-transform: scale(0.8) translate3d(0,-8px,0);
  -ms-transform: scale(0.8) translate3d(0,-8px,0);
  transform: scale(0.8) translate3d(0,-8px,0);
}

.c5:not(:placeholder-shown) {
  border-bottom-color: #64738C;
}

.c5:focus {
  border-bottom-color: #0DAFC0;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  margin-bottom: 35px;
  overflow: hidden;
}

.c10 {
  color: #64738C;
  font-size: 16px;
  font-family: "NunitoSans-500","sans-serif";
  margin-bottom: 10px;
  font-size: 13px;
}

.c11 {
  border: none;
  outline: none;
  border-bottom: 1px solid #64738C;
  padding-bottom: 8px;
  margin-left: -5px;
  width: calc(100% + 5px);
  color: #121212;
  font-size: 16px;
  font-family: "NunitoSans-600","sans-serif";
}

.c12 {
  width: 100%;
  height: 56px;
  border: none;
  border-radius: 12px;
  outline: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #fff;
  font-size: 14px;
  font-family: "NunitoSans-700","sans-serif";
  color: #fff;
  background-color: #E82223;
  cursor: default;
  background-color: #BBC3CF;
  color: #fff;
}

.c13 {
  background-color: #fafafa;
  width: 100%;
  height: 80px;
  position: fixed;
  bottom: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
}

.c2 {
  max-width: 600px;
  margin: 0 auto;
}

@media (min-width:576px) {
  .c1 {
    max-width: 540px;
  }
}

@media (min-width:768px) {
  .c1 {
    max-width: 720px;
  }
}

@media (min-width:992px) {
  .c1 {
    max-width: 960px;
  }
}

@media (min-width:1200px) {
  .c1 {
    max-width: 1140px;
  }
}

@media (min-width:1400px) {
  .c1 {
    max-width: 1320px;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <p
          class="c3"
          color="#121212"
          font-size="32"
          font-weight="600"
        >
          auth:signup-title
        </p>
        <form
          autocomplete="off"
        >
          <div
            class="c4"
          >
            <input
              class="c5"
              id="email"
              name="email"
              placeholder=" "
              type="email"
              value=""
            />
            <label
              class="c6 c7"
              for="email"
            >
              common:labels.email
            </label>
          </div>
          <div
            class="c8"
          >
            <input
              class="c5"
              id="password"
              name="password"
              placeholder=" "
              type="password"
              value=""
            />
            <label
              class="c6 c7"
              for="password"
            >
              common:labels.password
            </label>
          </div>
          <div
            class="c9"
          >
            <label
              class="c10"
              for="locale"
            >
              common:labels.locale
            </label>
            <select
              class="c11"
              data-testid="select"
              id="locale"
              name="locale"
            >
              <option
                data-testid="select-option"
                selected=""
                value="tr"
              >
                common:languages.tr
              </option>
              <option
                data-testid="select-option"
                value="en"
              >
                common:languages.en
              </option>
            </select>
          </div>
          <button
            class="c12"
            color="primary"
            disabled=""
            role="submit-button"
            type="submit"
          >
            auth:signup-button
          </button>
        </form>
      </div>
    </div>
    <div
      class="c13"
    >
      <a
        class="c14"
        data-testid="tab-list"
      >
        <svg
          color="#4D4D4D"
          data-testid="svg-icon"
          style="display: inline-block; stroke: currentColor; fill: currentColor;"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M568.889 0v341.333h455.111v-341.333h-455.111zM568.889 1024h455.111v-568.889h-455.111v568.889zM0 1024h455.111v-341.333h-455.111v341.333zM0 568.889h455.111v-568.889h-455.111v568.889z"
          />
        </svg>
        <p
          class="c15"
          color="#121212"
          font-size="12"
          font-weight="600"
        >
          common:tabbar.list
        </p>
      </a>
      <a
        class="c14"
        data-testid="tab-account"
      >
        <svg
          color="#4D4D4D"
          data-testid="svg-icon"
          style="display: inline-block; stroke: currentColor; fill: currentColor;"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M512 0c67.895 0 133.009 26.971 181.018 74.981 48.013 48.009 74.982 113.124 74.982 181.019s-26.97 133.010-74.982 181.018c-48.009 48.013-113.122 74.982-181.018 74.982s-133.010-26.97-181.019-74.982c-48.009-48.008-74.981-113.122-74.981-181.018s26.971-133.010 74.981-181.019c48.009-48.009 113.124-74.981 181.019-74.981zM512 640c282.88 0 512 114.56 512 256v128h-1024v-128c0-141.44 229.12-256 512-256z"
          />
        </svg>
        <p
          class="c15"
          color="#121212"
          font-size="12"
          font-weight="600"
        >
          common:tabbar.account
        </p>
      </a>
    </div>
  </div>
</div>
`;
